explainer.md

```diff
@@ -0,0 +1,259 @@
+  from {
+    opacity: 1;
+  }
+}
+::shared-old(#A) {
+  opacity: 0;
+  animation: fade-out;
+  object-fit: cover;
+  object-position: top center;
+}
+
+@keyframes fade-in {
+  from {
+    opacity:0;
+  }
+}
+::shared-old(#A) {
+  opacity: 1;
+  animation: fade-in;
+}
+
+// Easing curve/duration for change in element's transform and bounds.
+::shared-container(#A) {
+  transition: all 1s ease-out;
+}
+```
+
+For specifying animations using script, we will expose the [Animatable interface](https://drafts.csswg.org/web-animations/#the-animatable-interface-mixin) for the same pseudo elements.
+
+An API is also needed to customize the paint order of these elements.
+
+## Live Animatable Properties
+A common capability desirable for multiple use-cases during transitions was to interpolate styles like border-radius that change the element's shape. The [container transform](https://material.io/design/motion/the-motion-system.html#container-transform) example show a visual demo of that. Painting properties like the element's border within its image when using the element() function makes this difficult.
+
+Developers can control which properties are painted in the element vs interpolated by removing them from the shared element during the transition and adding them to the container instead. A new pseudo class is introduced which is applied when an element is in transition state, i.e., its being displayed in the Transition DOM. The following is an example of developer provided CSS to animate an element's border, border-radius and box-shadow during the transition. A working example of this using the existing element() function in Firefox is [here](https://jsbin.com/fifupusuvo/1/edit?html,output).
+
+Note : Supporting this requires the browser to copy the complete ComputedStyle for the pseudo elements in the Transition DOM from outgoing to incoming document, as opposed to only the [computed properties](#computed-properties).
+
+```
+.shared-element {
+  border: 10px solid black;
+  border-radius: 10% 10%;
+  box-shadow: 0px 0px 10px;
+  
+  &:transition {
+    /* Retain the border to ensure it is painted transparent but the box size is unchanged. */
+    border: 10px solid transparent;
+    border-radius: none;
+    box-shadow: none;
+  }
+}
+
+::shared-container(#shared-id) {
+  border: 10px solid black;
+  border-radius: 10% 10%;
+  box-shadow: 0px 0px 10px;
+}
+
+::shared-old(#shared-id) {
+  top: -10px;
+  left: -10px;
+}
+```
+
+An alternate approach to this is to support this natively in the browser by introducing a new content-element() function. This function would behave similar to the element() function except skipping the following properties when painting the element: box decorations and visual effects which generate a stacking context. The image will also be sized to the element's content-box (as opposed to the border-box used by the element() function). The motivation for supporting this natively would be to ensure ease of use for developers.
```

Comment:
You should clearly separate any alternatives into an "alternatives considered" section.

Reply:
Done.